src/vcpkg/postbuildlint.cpp

```diff
@@ -730,6 +730,47 @@ namespace vcpkg::PostBuildLint
         return LintStatus::SUCCESS;
     }
 
+    static LintStatus check_pkgconfig_dir_only_in_lib_dir(const Filesystem& fs, const Path& dir)
+    {
+        const auto allowed_in_share = [&fs](const Path& path) {
+            // If there is a "Lib:" or "Libs.private:" line in the .pc file, then the file must be in /lib/pkgconfig
+            return !Util::any_of(fs.read_lines(path, VCPKG_LINE_INFO),
+                                 [](const auto& line) { return Strings::starts_with(line, "Libs"); });
+        };
+
+        std::vector<Path> pkgconfig_files = fs.get_regular_files_recursive(dir, IgnoreErrors{});
+        Util::erase_remove_if(pkgconfig_files, [&allowed_in_share](const Path& path) {
+            if (!Strings::ends_with(path, ".pc")) return true;
+            const auto parent_path = Path(path.parent_path());
+            if (parent_path.filename() != "pkgconfig") return false;
+            const auto pkgconfig_parent_name = Path(parent_path.parent_path()).filename().to_string();
+            return pkgconfig_parent_name == "lib" || pkgconfig_parent_name == "manual-link" ||
+                   (pkgconfig_parent_name == "share" && allowed_in_share(path));
+        });
+
+        if (!pkgconfig_files.empty())
+        {
+            print2(Color::warning, "There should be no pkgconfig directories outside of lib and debug/lib.\n");
+            print2("The following misplaced pkgconfig files were found:\n");
+            print_paths(pkgconfig_files);
+            print2(
+                Color::warning,
+                "You can move the pkgconfig files with the following commands:\n"
+                "\n"
+                R"###(    file(INSTALL "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig/name.pc" DESTINATION "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig"))###"
+                "\n"
+                R"###(    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig"))###"
+                "\n"
+                "\nIf this check if a false positive you can disable it with `set(",
+                to_cmake_variable(BuildPolicy::SKIP_PKGCONFIG_LOCATION_CHECK),
+                " enabled)`"
```

Comment:
```suggestion
                "\nIf .pc files should be in this nonstandard location, you can use `set(",
                to_cmake_variable(BuildPolicy::SKIP_PKGCONFIG_LOCATION_CHECK),
                " enabled)` to suppress this message"
```

Reply:
(We can bikeshed this one :)

Reply:
The question is: Is there a use case where we would allow a .pc file outside of `lib(/debug)?(/manual-link)?/pkgconfig` or `share/pkgconfig`?

Reply:
I'm pretty sure they should always be there but I asked people who understand pkgconfig better than I to comment.

Reply:
Currently I can see no use cases -- in fact, we shouldn't allow them in `manual-link` either. I think restricting it to `lib/(debug/)?/pkgconfig` and `share/pkgconfig` is sufficient.

Reply:
agreed. I think we should remove the policy until we have a port that needs it.

Reply:
done